### PR TITLE
azfile: File download methods

### PR DESCRIPTION
**Pull request by souravgupta-msft**
This PR is built on the top of the file upload methods PR #20631.
Please review that PR #20631 before this.

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by vibhansa-msft — APPROVED** [review]

File: sdk/storage/azfile/directory/client_test.go [hunk 1]
```diff
@@ -219,63 +216,6 @@ func (d *DirectoryRecordedTestsSuite) TestDirectoryCreateNegativeMultiLevel() {
 	testcommon.ValidateFileErrorCode(_require, err, fileerror.ParentNotFound)
 }
 
-func (d *DirectoryUnrecordedTestsSuite) TestDirectoryClientUsingSAS() {
```
Comment (tasherif-msft) [inline sdk/storage/azfile/directory/client_test.go:222]: why are we deleting these, we can just have it as live only?

File: sdk/storage/azfile/file/client_test.go [hunk 2]
```diff
@@ -1408,12 +1601,1513 @@ func (f *FileRecordedTestsSuite) TestSASFileClientSignNegative() {
...
+
+	destFClient := testcommon.CreateNewFileFromShare(context.Background(), _require, "dest"+testcommon.GenerateFileName(testName), fileSize, shareClient)
+
+	_, err = destFClient.UploadRangeFromURL(context.Background(), srcFClient.URL(), 0, 0, int64(contentSize), &file.UploadRangeFromURLOptions{
+		SourceContentCRC64: contentCRC64,
+	})
+	_require.Error(err)
+}
+
+func (f *FileRecordedTestsSuite) TestFileUploadRangeFromURLOffsetNegative() {
```
Comment (tasherif-msft) [inline sdk/storage/azfile/file/client_test.go:1767]: You can set the base branch to be the upload pr/branch, so it doesnt show those changes and makes this pr more readable!

**Review by tasherif-msft — APPROVED** [review]

File: sdk/storage/azfile/directory/client_test.go (hunk 1 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/directory/client_test.go:222]: Service SAS at directory doesn't work. It works for share and file only.
https://learn.microsoft.com/en-us/rest/api/storageservices/create-service-sas#permissions-for-a-file

So, I have removed the SAS creation method in the directory client and also removed the tests related to it. If you check this test case, you can see that the service returns authentication error when directory SAS is used.

File: sdk/storage/azfile/file/client_test.go (hunk 2 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/file/client_test.go:1767]: Thanks for pointing this. The idea was to get the upload branch merged first before this PR goes for review.
But anyways I have the approvals for the upload PR and will merge it today.

**Pull request merged**